packages/mdc-checkbox/component.ts

```diff
@@ -0,0 +1,185 @@
+      setNativeControlDisabled: (disabled) => this.nativeCb_.disabled = disabled,
+    });
+  }
+
+  private initRipple_(): MDCRipple {
+    const foundation = new MDCRippleFoundation({
+      ...MDCRipple.createAdapter(this),
+      deregisterInteractionHandler: <K extends EventType>(evtType: K, handler: SpecificEventListener<K>) =>
+        this.nativeCb_.removeEventListener(evtType, handler),
+      isSurfaceActive: () => ponyfill.matches(this.nativeCb_, ':active'),
+      isUnbounded: () => true,
+      registerInteractionHandler: <K extends EventType>(evtType: K, handler: SpecificEventListener<K>) =>
+        this.nativeCb_.addEventListener(evtType, handler),
+    });
+    return new MDCRipple(this.root_, foundation);
+  }
+
+  private installPropertyChangeHooks_() {
+    const nativeCb = this.nativeCb_;
+    const cbProto = Object.getPrototypeOf(nativeCb);
+
+    CB_PROTO_PROPS.forEach((controlState) => {
+      const desc = Object.getOwnPropertyDescriptor(cbProto, controlState);
+      // We have to check for this descriptor, since some browsers (Safari) don't support its return.
+      // See: https://bugs.webkit.org/show_bug.cgi?id=49739
+      if (!validDescriptor(desc)) {
+        return;
+      }
+
+      const nativeCbDesc = {
+        configurable: desc.configurable,
+        enumerable: desc.enumerable,
+        get: desc.get,
+        set: (state: boolean) => {
+          desc.set!.call(nativeCb, state);
+          this.foundation_.handleChange();
+        },
+      };
+      Object.defineProperty(nativeCb, controlState, nativeCbDesc);
+    });
+  }
+
+  private uninstallPropertyChangeHooks_() {
+    const nativeCb = this.nativeCb_;
+    const cbProto = Object.getPrototypeOf(nativeCb);
+
+    CB_PROTO_PROPS.forEach((controlState) => {
+      const desc = Object.getOwnPropertyDescriptor(cbProto, controlState);
+      if (!validDescriptor(desc)) {
+        return;
+      }
+      Object.defineProperty(nativeCb, controlState, desc);
+    });
+  }
+
+  get ripple(): MDCRipple {
+    return this.ripple_;
+  }
+
+  get checked(): boolean {
+    return this.nativeCb_.checked;
+  }
+
+  set checked(checked: boolean) {
+    this.nativeCb_.checked = checked;
+  }
+
+  get indeterminate(): boolean {
+    return this.nativeCb_.indeterminate;
+  }
+
+  set indeterminate(indeterminate: boolean) {
+    this.nativeCb_.indeterminate = indeterminate;
+  }
+
+  get disabled(): boolean {
+    return this.nativeCb_.disabled;
+  }
+
+  set disabled(disabled: boolean) {
+    this.foundation_.setDisabled(disabled);
+  }
+
+  get value(): string {
+    return this.nativeCb_.value;
+  }
+
+  set value(value: string) {
+    this.nativeCb_.value = value;
+  }
+}
+
+function validDescriptor(inputPropDesc: PropertyDescriptor | undefined): inputPropDesc is PropertyDescriptor {
+  return !!inputPropDesc && typeof inputPropDesc.set === 'function';
+}
+
+export default MDCCheckbox;
```

Comment:
Remove all `default` exports from componentfiles. The existing JS code in `master` only has `default` exports for foundations.

Reply:
Yes, none of our existing JS components have default exports, so we shouldn't add them.